addon/components/ember-collection.js

```diff
@@ -80,17 +81,24 @@ export default Ember.Component.extend({
 
   updateItems(){
     this._cellLayout = this.getAttr('cell-layout');
-    var items = this.getAttr('items');
-
-    if (this._items !== items) {
-      if (this._items && this._items.removeObserver) {
-        this._items.removeObserver('[]', this, this._needsRevalidate);
+    var rawItems = this.getAttr('items');
+
+    if (this._rawItems !== rawItems) {
+      if (this._items && this._items.removeArrayObserver) {
+        this._items.removeEnumerableObserver(this, {
+          willChange: Ember.K,
+          didChange: '_needsRevalidate'
+        });
       }
-
+      this._rawItems = rawItems;
+      var items = Ember.A(rawItems);
       this.set('_items', items);
 
-      if (items && items.addObserver) {
-        items.addObserver('[]', this, this._needsRevalidate);
+      if (items && items.addEnumerableObserver) {
```

Comment:
With the previous code, calling addObserver with a property of `'[]'`raised on error when used with an ArrayProxy.